src/main/java/seedu/duke/util/Parser.java

```diff
@@ -248,12 +250,38 @@ private Command prepareListNote(String userMessage) {
 
     private Command prepareListEvent(String userMessage) {
         return new ListEventCommand();
-    }
+    }*/
 
     private Command prepareViewNote(String userMessage) {
-       return new ViewNoteCommand();
-    }
+        String title;
+        int index;
+        try {
+            ArrayList<String[]> splitInfo = splitInfoDetails(userMessage);
 
+            for (String[] infoDetails : splitInfo) {
+                String prefix = infoDetails[0];
+                if (infoDetails[1].isBlank()) {
+                    throw new SystemException(SystemException.ExceptionType.EXCEPTION_MISSING_DESCRIPTION);
+                }
+                switch (prefix) {
+                case PREFIX_TITLE:
+                    title = infoDetails[1].trim();
+                    return new ViewNoteCommand(title);
+                case PREFIX_INDEX:
+                    index = Integer.parseInt(infoDetails[1].trim());
+                    return new ViewNoteCommand(index);
+                default:
+                    return new IncorrectCommand("Missing tags!");
+                }
+            }
+        } catch (SystemException exception) {
+            return new IncorrectCommand(exception.getMessage());
+        } catch (ArrayIndexOutOfBoundsException | NullPointerException exception) {
+            return new IncorrectCommand("Missing description!");
+        }
+        return new IncorrectCommand("Incorrect format for viewing notes");
```

Comment:
There should not be any return new IncorrectCommand here as well as the catch block for SystemException. Can change the catch block for ArrayIndexOutBoundsException and NullPointerException to throw a new SystemException.

Suggest to split the catching of the two exceptions mentioned above to throw a more specific SystemException.